docs/API.md

```diff
@@ -51,6 +63,78 @@ HTMLReport
 XMLReport
 ```
 
+## Configuration
+
+Given you have the following configuration file called `config.reek` in your root directory:
+
+```Yaml
+---
+IrresponsibleModule:
+  enabled: false
+```
+
+You can now use either
+
+```Ruby
+Reek::Configuration::AppConfiguration.from_path Pathname.new('config.reek`)
+```
+
+but you can also pass a hash via `Reek::Configuration::AppConfiguration.from_map`.
+
+This hash can have the following 3 keys:
+
+1.) directory_directives [Hash] for instance:
+
+```Ruby
+  { Pathname("spec/samples/three_clean_files/") =>
+    { Reek::Smells::UtilityFunction => { "enabled" => false } } }
+```
+
+2.) default_directive [Hash] for instance:
+
+```Ruby
+  { Reek::Smells::IrresponsibleModule => { "enabled" => false } }
+```
+
+3.) excluded_paths [Array] for instance:
+
+```Ruby
+  [ Pathname('spec/samples/two_smelly_files') ]
+```
+
+Given the example above you should load that as "default directive" which means that it will
+be the default configuration for smell types for which there is
+no "directory directive" (so a directory-specific configuration):
+
+```Ruby
+require 'reek'
+
+default_directive = { Reek::Smells::IrresponsibleModule => { 'enabled' => false } }
+configuration = Reek::Configuration::AppConfiguration.from_map default_directive: default_directive
```

Comment:
Yeah, you're right, that's better indeed. However, this PR is a) concerned only with the API docs and (also in branch name) b) merely documenting already existing behaviour.
So how about merging this one and then updating "from_map" in a follow up PR so this doesn't get messy?
I created an issue for this so we don't forget about it:  https://github.com/troessner/reek/issues/662